cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -201,6 +207,30 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
             services[it.containerName] = service
         }
 
+        if (external.containerImage.isNotEmpty()) {
+            val extra = mutableMapOf(
+                    "image" to external.containerImage,
+                    "ports" to external.servicePorts.toList().map{QuotedString("${it}:${it}")},
+                    "expose" to external.servicePorts
+            )
+            if(external.volumes.isNotEmpty()){
+                extra["volumes"] = external.volumes.toList().map{ "${it["sourceFile"]}:${it["deploymentPath"]}" }
+            }
```

Comment:
The `volumes` property is a `ListProperty` and so has a default value of an empty list. This means that `.isPresent()` will always return `true`, and so you can do this:
```kotlin
val volumes = external.volumes.get()
if (volumes.isNotEmpty()) {
   ...
}
```
Again, you don't need `toList()` here.